tests/end_to_end/L1L2Messaging/test_messaging.py

```diff
@@ -55,9 +56,24 @@ async def l1_kakarot_messaging(sn_messaging_local, kakarot):
 
 
 @pytest_asyncio.fixture(scope="session")
-async def message_app_l2(owner):
+async def l2KakarotMessaging(kakarot):
+    l2KakarotMessaging = await deploy("L1L2Messaging", "L2KakarotMessaging")
+    await invoke(
+        "kakarot",
+        "set_authorized_cairo_precompile_caller",
+        int(l2KakarotMessaging.address, 16),
+        True,
+    )
+    return l2KakarotMessaging
+
+
+@pytest_asyncio.fixture(scope="session")
+async def message_app_l2(owner, l2KakarotMessaging):
     return await deploy(
-        "L1L2Messaging", "MessageAppL2", caller_eoa=owner.starknet_contract
+        "L1L2Messaging",
+        "MessageAppL2",
+        caller_eoa=owner.starknet_contract,
+        _l2KakarotMessaging=l2KakarotMessaging.address,
```

Comment:
```suggestion
        _l2KakarotMessaging=l2KakarotMessaging.address,
        caller_eoa=owner.starknet_contract,
```

both works, but `caller_eoa` is more while `_l2KakarotMessaging` is a constructor args, it's used in this order everywhere else